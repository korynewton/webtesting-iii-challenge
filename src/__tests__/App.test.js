import React from 'react';
import renderer from 'react-test-renderer';

import Dashboard from '../dashboard/Dashboard';

describe('Gate tests', () => {
  it('screenshots application', () => {
    const app = renderer.create(<Dashboard />).toJSON();
    expect(app).toMatchSnapshot();
  });
});

//ToDos:

//Gate:
// should default to unlocked and open
// cannot be closed or open if unlocked

//Dashboard:
// shows controls and display

//Display:
// displays if gate is open/closed and if it is locked/unlocked
// displays 'Closed' if the closed prop is true and 'Open' if otherwise
// displays 'Locked' if the locked prop is true and 'Unlocked' if otherwise
// when locked or closed use the red-led class
// when unlocked or open use the green-led class

//Controls Component:
// provide buttons to toggle the closed and locked states.
// buttons' text changes to reflect the state the door will be in if clicked
// the closed toggle button is disabled if the gate is closed
// the locked toggle button is disabled if the gate is open
