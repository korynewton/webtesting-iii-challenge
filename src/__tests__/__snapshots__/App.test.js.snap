// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gate tests screenshots application 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
